### DISCUSSION TITLE
ofetch - onOkResponse interceptor

### DISCUSSION THREAD

**Discussion by Amar-Gill** [nuxt/nuxt#24995] [Ideas]
Would anyone think it's helpful to add a new ofetch interceptor? https://github.com/unjs/ofetch#%EF%B8%8F-interceptors

I'm thinking about an `onOkResponse` interceptor.

Because in my project I have a custom `useFetch` call that has an `onResponseError` interceptor as well as an `onResponse` handler (for successful responses).

Reason is, I must wrap my logic in `onResponse` handler with `if (ctx.response.ok) { ... `. In the quest to eliminate boilerplate, I wonder if it would be valuable to add the `onOkResponse` interceptor, that only runs if `ctx.response.ok` is `true` - so the exact opposite of `onResponseError` which only runs if `ctx.response.error` is not `true`.

This idea stemmed from me debugging why my interceptor wouldn't run for a couple hours. The root cause was a silent error in `onResponse` which blocked `onResponseError` from running. (Maybe this should be a separate issue).

My `onResponse` handler depends on the request body to be defined, but during an error response it isn't defined and I got a silent error. Only after wrapping my logic with `if (ctx.response.ok)` does my `onResponseError` interceptor run.

**Comment by Amar-Gill** [comment]
btw here is a minimal reproduction of the interceptors getting blocked, I think it will work for any other set of interceptors:
```ts
const { data } = await useFetch('/api/has-error', {
  onResponse() {
    console.log('On response handler');
    throw new Error('todo');
  },
  onResponseError() {
    console.log('On response error handler');
  },
});
```

Since an error is thrown from `onResponse` -- `onResponseError` will not run.

**Comment by TheAlexLichter** [comment]
Best would be to open that discussion in https://github.com/unjs/ofetch (as an issue) ☺️

**Reply by Amar-Gill** [reply]
Makes sense! Here is a link to the issue:

https://github.com/unjs/ofetch/issues/348
